### PR TITLE
Improve TranslatableTrait

### LINKED ISSUE #5731: Extending Translatable resource
During development not always extending a resource should be connected with extending its class translation. But with a current implementation, we are forced to override [this method](https://github.com/Sylius/Sylius/blob/master/src/Sylius/Component/Resource/Model/TranslatableTrait.php#L161) or add extended class to the same namespace. We should think about some better solution for this.

### LINKED ISSUE #6739: [Admin][Product] Product->getSlug() tries to load non-existing translation class
I have a custom product model, extending the Product model from the CoreBundle, following the guidelines provided by the docs, http://docs.sylius.org/en/latest/customization/model.html.

It was working just fine until one of the slug related commits to the Product model was pushed. Now I am no longer able to create new products (editing existing works just fine), instead the following error is thrown:

stacktrace product model

I have not extended the ProductTranslation class, since I didn't need to, so I am not surprised it can't find that class...

As a workaround I have changed TranslatableTrait's row 163 from get_called_class() to get_class(), but I don't think that is a proper fix.

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets | fixes #6739, fixes #5731 |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Component/Resource/Model/TranslatableTrait.php [hunk 1]
```diff
@@ -48,30 +48,45 @@ public function __construct()
...
-    public function addTranslation(TranslationInterface $translation)
-    {
-        if (!$this->translations->containsKey($translation->getLocale())) {
-            $this->translations->set($translation->getLocale(), $translation);
-            $translation->setTranslatable($this);
+        if ($this->currentTranslation && $locale === $this->currentTranslation->getLocale()) {
+            return $this->currentTranslation;
+        }
+
+        $translation = $this->translations->get($locale)
```
Comment (NeverResponse) [inline src/Sylius/Component/Resource/Model/TranslatableTrait.php]: 🔭  missing `;` blows everything up

Comment (pamil) [inline src/Sylius/Component/Resource/Model/TranslatableTrait.php]: 💥

**Comment by michalmarcinkowski** [comment]
@pamil how about these two issues: https://github.com/Sylius/Sylius/issues/2695, https://github.com/Sylius/Sylius/issues/3562?

File: src/Sylius/Component/Resource/Model/TranslatableTrait.php [hunk 2]
```diff
@@ -85,81 +100,44 @@ public function hasTranslation(TranslationInterface $translation)
...
-     * Return translation model class.
-     *
-     * @return string
+     * @return TranslationInterface
      */
-    public static function getTranslationClass()
-    {
-        return get_called_class().'Translation';
-    }
+    abstract protected function createTranslation();
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Resource/Model/TranslatableTrait.php]: Comment with more detailed explanation for this method can be useful.

Comment (michalmarcinkowski) [inline src/Sylius/Component/Resource/Model/TranslatableTrait.php]: At least hint what to do in this 🦀  method

Comment (pamil) [inline src/Sylius/Component/Resource/Model/TranslatableTrait.php]: Done! :tada:

**Comment by pjedrzejewski** [comment]
Thanks Kamil!

**Pull request merged**